pkg/api/additional_metadata.go

```diff
@@ -3,9 +3,12 @@
 
 package api
 
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
 // +kubebuilder:object:generate=true
 
 type AdditionalMetadataSpec struct {
-	Labels      map[string]string `json:"labels,omitempty"`
-	Annotations map[string]string `json:"annotations,omitempty"`
+	NamespaceSelector *metav1.LabelSelector `json:"namespaceSelector,omitempty"`
+	Labels            map[string]string     `json:"labels,omitempty"`
+	Annotations       map[string]string     `json:"annotations,omitempty"`
 }
```

Comment:
Create a new spec for this, otherwise the other apis also have a nemspaceSelector field out of context.

```suggestion
	Labels            map[string]string     `json:"labels,omitempty"`
	Annotations       map[string]string     `json:"annotations,omitempty"`
}

type AdditionalMetadataSelectorSpec struct
	NamespaceSelector *metav1.LabelSelector `json:"namespaceSelector,omitempty"`
	Labels            map[string]string     `json:"labels,omitempty"`
	Annotations       map[string]string     `json:"annotations,omitempty"`
}

```

Reply:
Added the spec